app/controllers/concerns/account_recoverable.rb

```diff
@@ -1,5 +1,7 @@
 module AccountRecoverable
+  # :reek:FeatureEnvy
   def piv_cac_enabled_but_not_phone_enabled?
-    current_user.piv_cac_enabled? && !current_user.phone_enabled?
+    manager = TwoFactorAuthentication::MethodManager.new(current_user)
+    manager.two_factor_enabled?([:piv_cac]) && !manager.two_factor_enabled?(%i[sms voice])
```

Comment:
At a granular level, thoughts on keeping the communicative method names and moving the implementation details inside the manager class? This would be more readable:
```ruby
manager.piv_cac_enabled? && !manager.phone_enabled?
```
At a higher level, I'm not sure this is the right approach. Asking a method manager if it is `piv_cac_enabled?` doesn't sound right to me.

The way the `piv_cac_enabled_but_not_phone_enabled?` method is used is as a way to determine whether or not the user can access a certain page. That feels like a Policy Object to me, complemented with a Business Rule Object.

In a lot of our controllers, we have these `return redirect_to some_url if some_condition` and/or `before_action` callbacks that essentially determine if a user has satisfied certain business rules before they can access a page.

I don't think a name like `TwoFactorAuthentication::MethodManager` communicates the business rule that a PIV/CAC user must configure a backup phone number before they can use their account. Maybe a better name for `piv_cac_enabled_but_not_phone_enabled?` would be `piv_cac_user_has_not_configured_backup_phone_yet?`

For a good example of how to name a business rule object and methods, see https://www.rubytapas.com/2018/07/03/rule-object/

For general method naming best practices and Policy Objects, see the first 5 minutes of this great video: https://www.youtube.com/watch?v=0rsilnpU1DU

Reply:
Business rule objects will be a good next phase after the initial refactoring. I don't want to increase the scope of these PRs too much. This refactoring is making these patterns easier to see.

Specifically to the example here, our longer term goal is not to require a phone backup for piv/cac, but encourage people to have multiple 2fa options configured. With that in mind, I'm more inclined to add methods now to see if multiple 2fa options are configured (or, more generally, how many options are configured), and then modify behavior based on that. So rather than `piv_cac_enabled_but_not_phone_enabled?`, I'd want `user_needs_backup_2fa_method?` on a policy object that uses the `MethodManager` as a source of information.